GAME_DESIGN_PRINCIPLES.md

```diff
@@ -0,0 +1,122 @@
+
+### Depth in Simplicity
+Cubyz' special sauce is its simplicity; keeping everything simple on the surface makes the game approachable for beginners, while the hidden depth keeps it interesting for skilled players. Avoid features that would add unnecessary tedium to the player's experience. Always think about how it would effect the player's first impressions of the game.
+
+### Fuel the Player's Curiosity
+The world is filled with secrets, and we want the player to find these secrets on their own without any outside help or guide.
+
+### Problems have Multiple Solutions
+Problems faced in a particular age of progression should have multiple solutions, instead of just one solution for everything.
+For example, in the "Pre-Caves Age," the player will not have access to coal, as coal spawns low in the world and is shrouded in darkness. There's many solutions to this, however:
+- Find a dim, above ground light source.
+- Search for a cave with exposed coal.
+- Wander into the darkness to find a cave that's bright enough to mine in.
+- Find an above-ground structure that can grant the player coal.
+- Dig straight down until you find coal.
+
+Of course, this problem is completely solved as soon as the player gets coal, as they can now explore caves to find more coal with their newfound torches, but it's important that the player has these options in the first place to prevent them from getting stuck on a seemingly insurmountable barrier. This also gives them more freedom in how they can approach the game. Not one option is better or easier than the other, and if one turns out to be, then it should be allowed to be adjusted to encourage other methods.
+
+### Explain when Needed
+If something progress-related has no obvious explanation, the player will have to rely on a wiki to find out how to progress. One example is the bellows; players won't know to place it next to a furnace and jump on it repeatedly. The issue can be fixed by adding a tooltip that tells them how it works, while leaving out details they can intuitively find out themselves.
+
+Having a mechanic that isn't explained or illuded to makes it an "Invisible Mechanic." These can only be found out through looking at the game's code, wiki reading, or asking a developer, thus giving a disadvantage to casual players. It's best to avoid these at all costs.
+
+### Micro Moments
+These are tiny things the player does in between larger events; examples include:
+- Travelling
+- Parkour
+- Mining
+- Building
+- Crafting
+- Managing inventory
+- Fighting
+
+These moments are extremely important as they let the player mentally rest, so making sure they're as satisfying and consistent as possible is a must.
+
+### Make the World Feel Alive
+To add immersion to the game, creatures should perform behaviors outside of player input, such as hunting, migrating, eating, or sleeping.
```

Comment:
Very important point.
I've also thought about this a bit, and I've come to the conclusion that apart from these obvious behaviors, we also need a certain playful character as well to really get the immersion, e.g. how cats play with a mouse or how squirrels play catch with each other in the trees.

I guess  you can just add that to the list:
```suggestion
To add immersion to the game, creatures should perform behaviors outside of player input, such as hunting, playing, migrating, eating, or sleeping.
```